### PR TITLE
Fix discrepancy between tf.placeholder and tf.sparse.placeholder

### PR DESCRIPTION
**Pull request by yongtang**
When working on sparse placeholder I noticed that if shape is presented with `None` in any dims, the shape information is remvoed from sparse.placeholder. So `(None, 4)` will be `(None, None)`.

This is different from tf.placeholder:
```
root@ubuntu:/v# python
Python 2.7.15rc1 (default, Nov 12 2018, 14:31:15)
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
>>> v1 = tf.placeholder(tf.float32, (None, 4))
>>> v1.get_shape()
TensorShape([Dimension(None), Dimension(4)])
>>> v2 = tf.sparse.placeholder(tf.float32, (None, 4))
>>> v2.get_shape()
TensorShape([Dimension(None), Dimension(None)])
>>>
```

This fix adds the fix so that tf.placeholder and tf.sparse.placeholder
behaves the same.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow/python/ops/array_ops.py [hunk 1]
```diff
@@ -2085,7 +2085,13 @@ def _normalize_sparse_shape(shape, name):
...
   rank = shape.get_shape()[0] if isinstance(shape, ops.Tensor) else len(shape)
   if not isinstance(shape, ops.Tensor) and None in shape:
-    return (None, rank)
+    if not isinstance(shape, (list, tuple)):
+      return (None, rank)
+    # In case shape is a list or tuple, we will convert None into
+    # -1 (Unknwon dims) so that the shape is preserved when calling
+    # convert_to_tensor below. This will make sure tf.sparse.placeholder
+    # accepts `(None, 4)` like tf.placeholder.
+    shape = [dim if dim is not None else -1 for dim in shape]
```
Comment (alextp) [inline tensorflow/python/ops/array_ops.py]: I don't think putting -1s here is the right fix. Specifically, I worry that downstream users of that shape tensor will not be able to process the -1s in the elements.

My understanding of the shape tensor in sparse tensors is that, when evaluated, it should be the full shape of the sparse tensor, which this change breaks.

Can you add a longer test for the kind of code you want to support?

**Review by alextp — CHANGES_REQUESTED** [review]

**Comment by ymodak** [comment]
@yongtang gentle ping! Can you please address the changes requested? Thanks!

**Comment by tensorflowbutler** [comment]
It has been 15 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by tensorflowbutler** [comment]
It has been 43 days that this pull-request has stalled. Please create a new pull-request with the requested changes.

**Pull request closed without merging**